tools/Google.Cloud.Tools.TagReleases/Program.cs

```diff
@@ -45,70 +52,133 @@ private static int Main(string[] args)
+            var apis = ApiMetadata.LoadApis();
+            var newReleases = await ComputeNewReleasesAsync(client, apis);
+            ValidateChanges(newReleases);
 
-            if (!changed.Any())
+            if (!ConfirmReleases(apis, newReleases))
             {
-                Console.WriteLine("No releases need to be created. Exiting.");
-                return;
+                return 0;
             }
 
-            Console.WriteLine("Go ahead and create releases? (y/n)");
-            string response = Console.ReadLine();
-            bool createReleases = response == "y";
-            if (!createReleases)
+            await MakeReleasesAsync(client, newReleases, commit);
+            return 0;
+        }
+
+        private static async Task<string> FetchRemoteCommitAsync(GitHubClient client)
+        {
+            var remoteBranch = await client.Repository.Branch.Get(RepositoryOwner, RepositoryName, TargetBranch);
+            string sha = remoteBranch.Commit.Sha;
+            Console.WriteLine($"Current GitHub {TargetBranch} commit: {sha}");
+            return sha;
+        }
+
+        private static void ValidateLocalRepository(string expectedCommit)
+        {
+            var root = DirectoryLayout.DetermineRootDirectory();
+            using (var repo = new LibGit2Sharp.Repository(root))
             {
-                return;
+                string tip = repo.Head.Tip.Sha;
+                if (tip != expectedCommit)
+                {
+                    throw new UserErrorException($"Current local commit: {tip}. Aborting.");
+                }
+                var status = repo.RetrieveStatus("apis/apis.json");
+                if (status != FileStatus.Unaltered)
+                {
+                    throw new UserErrorException($"Expected apis.json to be unaltered. Current status: {status}. Aborting.");
+                }
             }
-            foreach (var api in changed)
+        }
+
+        private static async Task<List<ApiMetadata>> ComputeNewReleasesAsync(GitHubClient client, List<ApiMetadata> allApis)
+        {
+            var tags = (await client.Repository.GetAllTags(RepositoryOwner, RepositoryName)).Select(tag => tag.Name);
+            var noChange = allApis.Where(api => tags.Contains($"{api.Id}-{api.Version}") || api.Version.EndsWith("00")).ToList();
+            return allApis.Except(noChange).ToList();
+        }
+
+        /// <summary>
+        /// Validates the set of changes.
+        /// </summary>
+        /// <remarks>
+        /// <para>Current checks:</para>
+        /// <para>
+        /// Project references (in produuction code) are okay so long as all the targets of the references
+        /// are also going to be released. If this is not the case, the dependencies within the target could be different
+        /// to the ones in the public package version, causing a dependency issue in the package we're about to publish.
+        /// (This caused issue 1280 for example.)
```

Comment:
I don't think that'll do anything within the source code, to be honest... it's not like it's a github commit. Happy to do it anyway though.